### DISCUSSION TITLE
Disable warnEndValueEqualsCurrent on a per tween/sequence basis

### DISCUSSION THREAD

**Discussion by Nelvin** [KyryloKuzyk/PrimeTween#48] [General] [answered]
Hey,

recently started to use PrimeTween and really like it.

Is there a way to disable warnEndValueEqualsCurrent on a per tween basis?
I have a few situations where I do fade in/out effects combined with a small delay, and with fast mouse movements, chances are that some of the tweened values haven't changed in between.
Say, I do a fade in/out of a info panel when hovering over an object. The fade out uses a small delay so when I move the cursor away from and again over the object in a very short time, I get those warnings.

Disabling the warning just while setting up these tweens doesn't work as you probably do the check at a later point in time.

I could check if I need a fade at all, but it's sequences of fades, the code is much cleaner without this extra checks and I'd love to not disable the warning globally.

Thanks.

**Comment by KyryloKuzyk** [comment, marked as answer]
Hey,

Unfortunately, enabling the `warnEndValueEqualsCurrent` setting is impossible on a per-tween basis. I understand that checking the current value is not a perfect solution, but I'm afraid this is the only option for now:
```csharp
public Sequence Animate() {
    var sequence = Sequence.Create();
    if (image.color.a != targetAlpha) {
        sequence.Group(Tween.Alpha(image, targetAlpha, duration));
    }
    return sequence;
}
```

Yes, disabling the `warnEndValueEqualsCurrent` settings just before creating a tween and enabling the setting right after that will not work because PrimeTween doesn't start animation immediately, but waits for LateUpdate() (or for the startDelay as in your case).

One thing I have on my mind to address this issue is to save the `warnEndValueEqualsCurrent` inside a tween at the exact moment the tween is created. I should think about it more, thank you for the idea!

**Comment by KyryloKuzyk** [comment]
@Nelvin I implemented this feature request in version 1.1.12.

Now it's possible to disable the `warnEndValueEqualsCurrent` setting right before creating the tween (or many tweens), then enable it back immediately after that. This will disable the warning on all tweens that were created while the `warnEndValueEqualsCurrent` was disabled.

**Reply by Nelvin** [reply]
@KyryloKuzyk Wow, that's great. Didn't expect you to make this change this fast.
Updated the package and gave it a try - perfect!
Thanks a lot.
